Source/DafnyCore/Coco/Parser.frame

```diff
@@ -178,20 +191,52 @@ public class Errors {
   public void Deprecated(IToken tok, string msg) {
     Contract.Requires(tok != null);
     Contract.Requires(msg != null);
-    Reporting.Deprecated(MessageSource.Parser, tok, msg);
+    Reporting.Deprecated(MessageSource.Parser, "", tok, msg);
   }
 
-  public void DeprecatedStyle(IToken tok, string msg) {
+  public void DeprecatedStyle(string errorID, IToken tok, string msg) {
     Contract.Requires(tok != null);
     Contract.Requires(msg != null);
-    Reporting.DeprecatedStyle(MessageSource.Parser, tok, msg);
+    Reporting.DeprecatedStyle(MessageSource.Parser, errorID, tok, msg);
   }
 
   public void Warning(IToken tok, string msg) {
     Contract.Requires(tok != null);
     Contract.Requires(msg != null);
     Reporting.Warning(MessageSource.Parser, tok, msg);
   }
+
+  // This is a RangeToken with the given length, but is a CodeActionToken
+  // so that it can be distinguished later
+  public static CodeActionToken NewCodeAction(IToken t, int len) {
+    Token x = (Token)t;
+    var y = new Token(x.line,x.col);
+    y.pos = x.pos;
+    y.kind = x.kind;
+    y.Filename = x.Filename;
+    y.val = len <= x.val.Length ? x.val.Substring(0, len) : x.val.PadRight(len, ' ');
+    return new CodeActionToken(t, y);
+  }
+
+  // Nested range is up to, not including, the next token
+  public static IToken NestCodeAction(IToken t, IToken next) {
```

Comment:
```suggestion
  public static IToken NestCodeAction(IToken t, IToken next = null) {
    next = next ?? t.Next;
```
That way, when you know that the second token is the next, you don't need to repeat the token in the arguments.